src/v/cloud_storage/cache_service.cc

```diff
@@ -1970,4 +1970,27 @@ ss::future<> cache::sync_access_time_tracker(
     }
 }
 
+std::optional<ss::sstring>
+cache::validate_cache_config(const config::configuration& conf) {
+    const auto& cloud_storage_cache_size = conf.cloud_storage_cache_size;
+    const auto& cloud_storage_cache_size_pct
+      = conf.cloud_storage_cache_size_percent;
+
+    // If not set, cloud cache uses default value of 0.0
+    auto cache_size_pct = cloud_storage_cache_size_pct().value_or(0.0);
+
+    using cache_size_pct_type = double;
+    static constexpr auto epsilon
+      = std::numeric_limits<cache_size_pct_type>::epsilon();
+
+    if ((cache_size_pct < epsilon) && (cloud_storage_cache_size() == 0)) {
```

Comment:
please add a commit that adds a comment explaining this. i read the docs for epsilon, but i don't see how it fits into this calculation

Reply:
For `double` floating-point comparison to 0- is there a different way you prefer doing this?

I can change the commit history to be clear.

Reply:
>I can update the commit history to be clear.
this already merged, so i think it'll have to be a comment?